parsl/channels/ssh/ssh.py

```diff
@@ -119,7 +128,7 @@ def execute_wait(self, cmd, walltime=2, envs={}):
         '''
 
         # Execute the command
-        stdin, stdout, stderr = self.ssh_client.exec_command(
+        stdin, stdout, stderr = self._connection()[0].exec_command(
```

Comment:
I think this _connection()[0] style is really ugly, compared to the previously style. It is less clear than before what kind of connection is being used here. It is especially disappointing because there are two nicely named fields that _connection is maintaining (self.ssh_client, and self.sftp_client).

A nicer way might be to make two accessor functions, `self.valid_ssh_client()` and `self.valid_sftp_client()` which are something like this:

  def valid_sftp_client():
    self._connection()
    return self.sftp_client

That is, they check for the connection being valid, before returning that appropriate connection.

Then the above line would look like:
```
stdin, stdout, stderr = self.valid_ssh_client().exec_command(
```